src/esm_parser/provenance.py

```diff
@@ -130,18 +134,27 @@ def get_provenance(self, dictionary=None):
         """
 
         provenance_dict = {}
-        if not dictionary:
+        if dictionary is None:
             dictionary = self
 
         for key, val in dictionary.items():
             if isinstance(val, dict):
                 provenance_dict[key] = self.get_provenance(val)
+            elif hasattr(dictionary, "provenance"):
+                provenance_dict[key] = dictionary.provenance.get(key)
             else:
-                provenance_dict[key] = dictionary.provenance[key]
+                provenance_dict[key] = None
```

Comment:
I'm not sure I understand this case. A comment would be helpful for future maintenance